src/moonlink/src/storage/wal.rs

```diff
@@ -404,6 +437,67 @@ impl WalManager {
         Ok(())
     }
 
+    /// Persist a series of wal events to the file system.
+    /// Should be called asynchronously using the results of take_for_next_file.
+    pub async fn persist(
+        file_system_accessor: Arc<dyn BaseFileSystemAccess>,
+        wal_to_persist: &Vec<WalEvent>,
+        wal_file_info: &WalFileInfo,
+    ) -> Result<()> {
+        if !wal_to_persist.is_empty() {
+            let wal_json = serde_json::to_vec(&wal_to_persist)?;
+
+            let wal_file_path = WalManager::get_file_name(wal_file_info.file_number);
+            file_system_accessor
+                .write_object(&wal_file_path, wal_json)
+                .await?;
+        }
+        Ok(())
+    }
+
+    pub async fn wal_persist_truncate_async(
+        files_to_delete: Vec<WalFileInfo>,
+        file_to_persist: Option<(Vec<WalEvent>, WalFileInfo)>,
+        file_system_accessor: Arc<dyn BaseFileSystemAccess>,
+        table_notify: Sender<TableEvent>,
+        iceberg_snapshot_lsn: Option<u64>,
+    ) {
+        // Execute WAL operations
+        let result = async {
+            let file_system_accessor_persist = file_system_accessor.clone();
+
+            // Delete old WAL files
+            if !files_to_delete.is_empty() {
+                WalManager::delete_files(file_system_accessor, &files_to_delete).await?;
+            }
+
+            // Persist new WAL file
+            if let Some((wal_events, wal_file_info)) = &file_to_persist {
+                WalManager::persist(file_system_accessor_persist, wal_events, wal_file_info)
```

Comment:
the delete all files implementation should be placed into filesystem layer, instead of application layer